pkg/querier/queryrange/roundtrip.go

```diff
@@ -761,19 +775,71 @@ func NewMetricTripperware(cfg Config, engineOpts logql.EngineOpts, log log.Logge
 }
 
 // NewInstantMetricTripperware creates a new frontend tripperware responsible for handling metric queries
-func NewInstantMetricTripperware(cfg Config, engineOpts logql.EngineOpts, log log.Logger, limits Limits, schema config.SchemaConfig, metrics *Metrics, indexStatsTripperware base.Middleware, metricsNamespace string) (base.Middleware, error) {
+func NewInstantMetricTripperware(
+	cfg Config,
+	engineOpts logql.EngineOpts,
+	log log.Logger,
+	limits Limits,
+	schema config.SchemaConfig,
+	metrics *Metrics,
+	merger base.Merger,
+	c cache.Cache,
+	cacheGenNumLoader base.CacheGenNumberLoader,
+	retentionEnabled bool,
+	indexStatsTripperware base.Middleware,
+	metricsNamespace string,
+) (base.Middleware, error) {
+	var cacheMiddleware base.Middleware
+	if cfg.CacheInstantMetricResults {
```

Comment:
Generally prefer the happy path to be as unindented as possible; so you could invert this `if` and exit early.

Reply:
true. But I think it applies only if we **could** just return and exit without needing any additional steps. 

Here in `NewInstantMetricTripperware` we do some additional steps in either case where `cfg.CacheInstantMetricResults` is true or false.

If I invert, I would still need more `if` to create other middlewares. Basically we cannot just return or exist directly if it is unhappy path

Did that answer your concern or am I failing to understand something else?